geom/alg_convex_hull.go

```diff
@@ -158,22 +144,21 @@ func sortByPolarAngle(ps []XY) {
 	// the lowest-then-leftmost (anchor) point comes first
 	ltlp := ltl(ps)
 	ps[ltlp], ps[0] = ps[0], ps[ltlp]
-	anchor := ps[0]
+	virtualPoint := ps[0]
 
 	ps = ps[1:] // only sort the remaining points
-	sort.Slice(ps, func(i, j int) bool {
-		// If any point is equal to the anchor point, then always put it first.
-		// This allows those duplicated points to be removed when the results
-		// stack is initiated.
-		if anchor.Equals(ps[i]) {
-			return true
-		}
-		if anchor.Equals(ps[j]) {
+	sort.SliceStable(ps, func(i, j int) bool {
```

Comment:
I recall that there was a problem using plain `sort.Slice` rather than `sort.SliceStable`. I'm not convinced that using a stable sort actually makes sense here, because the inputs can appear in any order (and the ordering of the inputs doesn't have any meaning). I'd prefer that we get to the bottom of _why_ `sort.Slice` wasn't working.